src/components/Tooltip/Tooltip.tsx

```diff
@@ -1,140 +1,121 @@
+import Tippy from '@tippyjs/react';
 import clsx from 'clsx';
-import React, {
-  ReactNode,
-  useState,
-  useRef,
-  useEffect,
-  MutableRefObject,
-} from 'react';
+import * as React from 'react';
 import styles from './Tooltip.module.css';
-import { ESCAPE_KEYCODE, TAB_KEYCODE } from '../../util/keycodes';
-import { Button } from '../Button/Button';
 
-export interface Props {
+// Full list of Tippy props: https://atomiks.github.io/tippyjs/v6/all-props/
+type TooltipProps = {
   /**
-   * Alignment variations
-   * - **right** yields a tooltip that appears to the right of the trigger
-   * - **below** yields a tooltip that appears below the trigger
-   * - **left** yields a tooltip that appears to the left of the trigger
+   * The trigger element the tooltip appears next to.
    */
-  align?: 'right' | 'below' | 'left';
+  children?: React.ReactElement;
   /**
-   * The visually-hidden button text for the tooltip trigger button
+   * The trigger element the tooltip appears next to.
+   *
+   * Use this instead of `children` if the trigger element is being
+   * stored in a ref. Most cases will use `children` and not
+   * `reference`.
    */
-  buttonText?: string;
+  reference?: React.RefObject<Element> | Element;
   /**
-   * Child node(s) that can be nested inside component. This gets displayed inside a `TextPassage` inside the tooltip
+   * The content of the tooltip bubble.
    */
-  children?: ReactNode;
+  content?: React.ReactNode;
   /**
-   * CSS class names that can be appended to the component.
+   * Where the tooltip should be placed in relation to the element it's attached to.
+   *
+   * Tippy also supports 'top-start', 'top-end', 'right-start', 'right-end', etc,
+   * but our CSS currently only supports the 4 main sides.
+   */
+  align?: 'top' | 'right' | 'bottom' | 'left';
+  /**
+   * Whether the tooltip is always visible or always invisible.
+   *
+   * This is most often left undefined so the Tooltip component
+   * controls if/when the bubble appears (on hover, click, focus, etc).
+   */
+  visible?: boolean;
+  /**
+   * Custom classname for additional styles.
+   *
+   * These styles will only affect the tooltip bubble.
    */
   className?: string;
   /**
-   * HTML id for the component
+   * Whether the tooltip has a light or dark background.
    */
-  id?: string;
-}
-
-/**
- * Primary UI component for user interaction
- */
-export const Tooltip: React.FC<Props> = ({
-  className,
-  children,
-  align,
-  buttonText,
-  ...other
-}) => {
+  variant?: 'light' | 'dark';
```

Comment:
Should this be `inverted`?

Reply:
I'm not sure 🤔 `inverted` is when the color scheme is inverted so it will have enough contrast on a dark background, but both of these variants work on light and dark backgrounds. Actually, this might be a good time to bring up whether we really need both of them, and, if we do, what are the guidelines around when to use which?

Reply:
yeah that's a great point @dierat

Reply:
Flagged this conundrum with the team over here and we're hopefully helping make some calls here: https://bigmedium.slack.com/archives/C0310HJ9WEN/p1646949365214339